infra/Pos/Reporting/Methods.hs

```diff
@@ -118,23 +117,20 @@ ipv4Local w =
 
 -- | Retrieves node info that we would like to know when analyzing
 -- malicious behavior of node.
-getNodeInfo :: (MonadDHT m, MonadIO m) => m Text
+getNodeInfo :: (MonadIO m) => m Text
```

Comment:
Many definitions use `reportingFatal`, and `reportingFatal` needs to know the peers because it prints them. Building-in exception handling in this way is not ideal. It means the handler can know only as much as the exception-throwing term itself. This implies that if the exception *handler* needs discovery, then the exception-throwing term must also have discovery.

There's a better way: move uses of `reportingFatal` up higher, to where the discovery obligation can be discharged.